pkg/pillar/cmd/zedrouter/pubsubhandlers.go

```diff
@@ -629,3 +629,134 @@ func (z *zedrouter) handleAppInstDelete(ctxArg interface{}, key string,
 	z.unpublishAppInstMetadata(appInstMetadata)
 	z.log.Functionf("handleAppInstDelete(%s) done", key)
 }
+
+func (z *zedrouter) handlePatchEnvelopeImpl(peInfo types.PatchEnvelopeInfoList) {
+	z.patchEnvelopes.UpdateEnvelopes(peInfo.Envelopes)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
```

Comment:
No need for a goroutine. Just do a conditional send on the channel. See e.g., the 5 lines at https://github.com/lf-edge/eve/blob/0f7bbc45eaf1e5cda92181c706f345384c648b18/pkg/pillar/cmd/downloader/handlers.go#L54